### PR TITLE
tk: add tk/8.6.10 recipe

### LINKED ISSUE #3449: [request] tk/8.6.10
### Package Details
  * Package Name/Version: **tk/8.6.10**
  * Website: **http://tcl.tk/**
  * Source code: **https://github.com/tcltk/tk**

### Description Of The Library / Tool
Tk is a free and open-source, cross-platform widget toolkit that provides a library of basic elements of GUI widgets for building a graphical user interface (GUI) in many programming languages.

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **tk/8.6.10**

Fixes #3449
Required by #1510

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries @SSE4 
@prince-chrismc @Croydon @prince-chrismc 

Can I please get some reviews/feedback on this pr?

File: recipes/tk/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+    def _get_configure_folder(self, build_system=None):
+        if build_system is None:
+            build_system = self._get_default_build_system()
+        if build_system not in ["win", "unix", "macosx"]:
+            raise ConanExceptionInUserConanfileMethod("Invalid build system: {}".format(build_system))
+        return os.path.join(self.source_folder, self._source_subfolder, build_system)
+
+    def _build_nmake(self, target="release"):
+        # https://core.tcl.tk/tips/doc/trunk/tip/477.md
```
Comment (prince-chrismc) [inline recipes/tk/all/conanfile.py:111]: Should there be a check for windows and/or msvc in this function?

File: recipes/tk/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        else:
+            with tools.chdir(self.build_folder):
+                autotools, make_args = self._configure_autotools()
+                autotools.install(args=make_args)
+                autotools.make(target="install-private-headers", args=make_args)
+                tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "man"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+        # FIXME: move to patch
```
Comment (prince-chrismc) [inline recipes/tk/all/conanfile.py:202]: ⏰ Sounds reasonable to do in his PR

**Review by prince-chrismc — APPROVED** [review]
I have no objections, it works so I did not verify the patch sources but they sound reasonable wiht good comments

File: recipes/tk/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/tk/all/conanfile.py:111]: The compiler is checked before calling this method (in `build` and `package`).

I think, in theory, nmake can build with gcc.
Nobody sane would do that.
But it's possible.

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**